src/devices/RGBDSensorWrapper/rgbdSensor_nws_yarp.cpp

```diff
@@ -0,0 +1,665 @@
+        yCError(RGBDSENSORNWSYARP, "Opening IRGBDSensor subdevice... FAILED");
+        return false;
+    }
+    isSubdeviceOwned = true;
+    if(!attach(subDeviceOwned)) {
+        return false;
+    }
+
+    return true;
+}
+
+bool RgbdSensor_nws_yarp::close()
+{
+    yCTrace(RGBDSENSORNWSYARP, "Close");
+    detachAll();
+
+    // close subdevice if it was created inside the open (--subdevice option)
+    if(isSubdeviceOwned)
+    {
+        if(subDeviceOwned)
+        {
+            delete subDeviceOwned;
+            subDeviceOwned=nullptr;
+        }
+        sensor_p = nullptr;
+        fgCtrl = nullptr;
+        isSubdeviceOwned = false;
+    }
+
+    // Closing port
+
+    rpcPort.interrupt();
+    colorFrame_StreamingPort.interrupt();
+    depthFrame_StreamingPort.interrupt();
+
+    rpcPort.close();
+    colorFrame_StreamingPort.close();
+    depthFrame_StreamingPort.close();
+
+    return true;
+}
+
+/* Helper functions */
+
+bool RgbdSensor_nws_yarp::initialize_YARP(yarp::os::Searchable &params)
+{
+    // Open ports
+    bool bRet;
+    bRet = true;
+    if(!rpcPort.open(rpcPort_Name))
+    {
+        yCError(RGBDSENSORNWSYARP) << "Unable to open rpc Port" << rpcPort_Name.c_str();
+        bRet = false;
+    }
+    rpcPort.setReader(rgbdParser);
+
+    if(!colorFrame_StreamingPort.open(colorFrame_StreamingPort_Name))
+    {
+        yCError(RGBDSENSORNWSYARP) << "Unable to open color streaming Port" << colorFrame_StreamingPort_Name.c_str();
+        bRet = false;
+    }
+    if(!depthFrame_StreamingPort.open(depthFrame_StreamingPort_Name))
+    {
+        yCError(RGBDSENSORNWSYARP) << "Unable to open depth streaming Port" << depthFrame_StreamingPort_Name.c_str();
+        bRet = false;
+    }
+
+    return bRet;
+}
+
+void RgbdSensor_nws_yarp::setId(const std::string &id)
+{
+    sensorId=id;
+}
+
+std::string RgbdSensor_nws_yarp::getId()
+{
+    return sensorId;
+}
+
+/**
+  * IWrapper and IMultipleWrapper interfaces
+  */
+bool RgbdSensor_nws_yarp::attachAll(const PolyDriverList &device2attach)
+{
+    // First implementation only accepts devices with both the interfaces Framegrabber and IDepthSensor,
+    // on a second version maybe two different devices could be accepted, one for each interface.
+    // Yet to be defined which and how parameters shall be used in this case ... using the name of the
+    // interface to view could be a good initial guess.
+    if (device2attach.size() != 1)
+    {
+        yCError(RGBDSENSORNWSYARP, "Cannot attach more than one device");
+        return false;
+    }
+
+    yarp::dev::PolyDriver * Idevice2attach = device2attach[0]->poly;
+    if(device2attach[0]->key == "IRGBDSensor")
+    {
+        yCInfo(RGBDSENSORNWSYARP) << "Good name!";
+    }
+    else
+    {
+        yCInfo(RGBDSENSORNWSYARP) << "Bad name!";
+    }
+
+    if (!Idevice2attach->isValid())
+    {
+        yCError(RGBDSENSORNWSYARP) << "Device " << device2attach[0]->key << " to attach to is not valid ... cannot proceed";
```

Comment:
Just a tiny remark: YARP stream logging utilities already take care of adding spaces surrounding each `<<` argument, therefore the last space in the "Device" string and the first one in "to attach..." would be duplicated. Not sure if intentional, it's written this way in several other places throughout the code.